Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -24,6 +28,36 @@ public let appReducer: Reducer<
         action: /AppAction.main,
         environment: \AppEnvironment.main
     ),
+    Reducer { state, action, environment in
+        switch action {
+        case let .didLogIn(jid):
+            state.route = .main(MainScreenState(
+                sidebar: .init(credentials: .init(jid: jid))
+            ))
+
+        case let .auth(.loginResult(.success(jid))):
+            if let url = URL(string: "prose://main") {
```

Comment:
No need to if/let here. Let's crash (see below for more info).

Here's a helper to isolate the force-unwrapping:

```swift
public extension Optional {
  /// Either returns the unwrapped value or prints `message` and stops execution of the program.
  func expect(_ message: String) -> Wrapped {
    guard let value = self else {
      fatalError(message)
    }
    return value
  }
}
```

Reply:
I usually have a `URL.init(staticString:) → URL` initializer, but I'll add your more general purpose function.

Reply:
That sounds even better actually (staticString).

Reply:
So are you OK with this?

```swift
import Foundation

extension URL {
	/// Comes from [Constructing URLs in Swift](https://www.swiftbysundell.com/articles/constructing-urls-in-swift/#static-urls)
	public init(staticString string: StaticString) {
		guard let url = URL(string: "\(string)") else {
			preconditionFailure("Invalid static URL string: \(string)")
		}
		self = url
	}
}
```

Reply:
Yes, that's great!